### PR TITLE
feature/syslog-parser-tenant-aware

### LINKED ISSUE #5549: Syslog parsers aren't tenant aware
**Describe the bug**
The syslog parsers (pfdetect) aren't tenant aware which makes it impossible to trigger a security event for a node of a specific tenant

**To Reproduce**
1. Setup multi-tenancy
2. Setup a suricata syslog parser
3. Send an alert `echo "Apr 18 10:48:17 pfdetect(6716) INFO: alert received: 04/18/2018-10:48:17.643084 [**] [1:2002878:7] ET POLICY iTunes User Agent [**] [Classification: Potential Corporate Privacy Violation] [Priority: 1] {TCP} 172.20.20.169:52801 -> 165.254.0.105:80" > /usr/local/pf/var/run/suricata_alert`
4. The alert is always triggered for tenant 1

**Expected behavior**
We should have a tenant parameter for the syslog parsers to trigger events for different tenants

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Make syslog parser tenant aware

NEWS file entries
=======================

Enhancements
-------------
* Syslog parsers are now tenant aware.

Issue
=====
fixes #5549

Delete branch after merge
=========================
NO

Checklist
=========
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

### CODE REVIEW

File: go/detectparser/parser.go [hunk 1]
```diff
@@ -44,6 +45,22 @@ type RateLimitable struct {
...
+}
+
+func setupParser(config *PfdetectConfig) parser {
+	if config == nil {
+		return parser{RateLimitable: RateLimitable{}, TenantID: 1}
+	}
+
+	return parser{
+		RateLimitable: NewRateLimitable(config.RateLimit),
+		TenantID:      config.TenantID,
```
Comment (julsemaan) [inline go/detectparser/parser.go:60]: We should ship an upgrade script to ensure tenant_id will always be set in the configuration

File: go/pfqueueclient/pfqueueclient.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ func (c *PfQueueClient) SubmitWithExpiration(ctx context.Context, queue, task_ty
 	}
 
 	redisClient.PipeAppend("MULTI")
-	redisClient.PipeAppend("HMSET", id, "expire", expire_in, "data", data)
+	redisClient.PipeAppend("HMSET", id, "expire", expire_in, "data", data, "tenant_id", tenant_id)
```
Comment (julsemaan) [inline go/pfqueueclient/pfqueueclient.go:56]: Can you confirm that there is existing magic to set the tenant ID in the DAL and config in pfqueue?
I don't see any changes related to that so I just want to be sure its already there.

File: html/pfappserver/lib/pfappserver/Form/Config/Pfdetect.pm [hunk 3]
```diff
@@ -58,11 +59,28 @@ has_field 'type' =>
    required => 1,
   );
 
+has_field 'tenant_id' =>
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Pfdetect.pm:62]: This field is missing from the frontend

**Review by julsemaan — CHANGES_REQUESTED** [review]

File: go/pfqueueclient/pfqueueclient.go (hunk 2 above)
Comment (jrouzierinverse) [reply go/pfqueueclient/pfqueueclient.go:56]: It has been there for a long time look at the commit a998fff4.

File: go/detectparser/parser.go (hunk 1 above)
Comment (jrouzierinverse) [reply go/detectparser/parser.go:60]: ok

**Comment by jrouzierinverse** [comment]
Please re-review.

**Comment by extrafu** [comment]
@julsemaan Final review and merge please.

**Pull request merged**